src/Swashbuckle.AspNetCore.SwaggerGen/DependencyInjection/SwaggerGenServiceCollectionExtensions.cs

```diff
@@ -33,8 +32,16 @@ public static class SwaggerGenServiceCollectionExtensions
             services.TryAddTransient<ISerializerDataContractResolver>(s =>
             {
 #if (!NETSTANDARD2_0)
-                var serializerOptions = s.GetService<IOptions<JsonOptions>>()?.Value?.JsonSerializerOptions
+                var serializerOptions =
+                    s.GetService<IOptions<AspNetCore.Mvc.JsonOptions>>()?.Value?.JsonSerializerOptions
+#if NET8_0_OR_GREATER
+                    ?? s.GetService<IOptions<AspNetCore.Http.Json.JsonOptions>>()?.Value?.SerializerOptions
+#endif
+#if NET7_0_OR_GREATER
+                    ?? JsonSerializerOptions.Default;
```

Comment:
Would it be sensible / useful  to be able to pass in an instance of `JsonSerializerOptions`.

Reply:
Possibly.